### DISCUSSION TITLE
tp.system.suggester choose from a set of files to be used when creating templates

### DISCUSSION THREAD

**Discussion by biopixen** [SilentVoid13/Templater#993] [Help] [answered]
Hi! I have spent several hours trying to read forum posts, watching youtube videos and are now stuck.

**In short what i would like to do:**
When i create a new note (using templater) i have several questions that need manual input. One of the questions are:
_Head MOC file?_
This name/variable is then used in the template as <%MOCNAME%> in order to directly link the note to the head MOC-file.

Example:
I have a MOC note called "MOC - Projects"
I create a new note and manually write _MOC - Projects_ in the "What is the HEAD MOC File?"prompt/question. _(Now saved as <%aMOCNAME%>)_ 
The note then contains a [[<%aMOCNAME%>]] link and when the note is created there is a OUTLINK to the MOC-Document.

Ok.
So, i have figured out how to search for and list all my notes that contains MOC
```
dataview
LIST FROM "" WHERE contains(file.name, "MOC") 
```
I also found a codesnippet:
 
`(await tp.system.suggester((item) => item.basename, app.vault.getMarkdownFiles().filter(file => file.path.startsWith("SANDBOX")), false, "Please link previous sprint plan")).basename
`

That prompts me to choose from a list of files.

How could i combine these two to create templater-code that asks me to choose the HEAD MOC file and then use that name to populate the [[]] link.

Im sorry if this is the wrong forum or weirdly stated question.

**Comment by dmych** [comment, marked as answer]
Hi, if I get it right, you could use something like this:
```
[[<% (await tp.system.suggester((item) => item.basename, app.vault.getMarkdownFiles().filter(file => file.path.includes("MOC")), false, "What is the HEAD MOC File?")).basename %>]]
```
This template asks user to choose one of files containing "MOC" in the file name and inserts internal link to the file.

**Reply by biopixen** [reply]
Wonderful! Thank you so much! Worked perfectly :)

**Reply by biopixen** [reply]
2 follow up questions:
* Is it possible to do the same thing but search for any files that includes the TAG #MOC as instead? Ignoring filename?
* Is it possible to store the chosen file as a variable for me to fill in frontend information: alias: <%chosenfile%> etc.

**Reply by dmych** [reply]
1. Sure, it is possible somehow. May be using `vault.cachedRead()` or `vault.read()` [functions](https://marcus.se.net/obsidian-plugin-docs/vault#read-files) to get content of every file and then searching in it. But I'm not a Javascript guru. :)
2. Yes, of couse, since the template inside `<% ... %>` is actually Javascript code. You can try something similar to:
```
[[<%*
let answer = (await tp.system.suggester((item) => item.basename, app.vault.getMarkdownFiles().filter(file => file.path.includes("MOC")), false, "What is the HEAD MOC File?")).basename
tR += answer %>]]
<%
tp.file.include('[['+answer+']]')
%>
```
Notice, that the basename of the choosen file stored in variable `answer` and then used twice: to add to the text via `tR` (see [here for details](https://silentvoid13.github.io/Templater/commands/execution-command.html#how-to-output-a-value-from-a-javascript-execution-command-)); to include template using `tp.file.include()`.

**Reply by biopixen** [reply]
Worked perfectly! Thank you again! ill try and see if i can find a solution for #1
Again, thank you so much! :)
